userspace/libpman/src/stats.c

```diff
@@ -234,17 +238,57 @@ struct metrics_v2 *pman_get_metrics_v2(uint32_t flags, uint32_t *nstats, int32_t
 			g_state.stats[MODERN_BPF_N_DROPS_SCRATCH_MAP].value.u64 += cnt_map.n_drops_max_event_size;
 			g_state.stats[MODERN_BPF_N_DROPS].value.u64 += (cnt_map.n_drops_buffer + cnt_map.n_drops_max_event_size);
 
+			if((flags & METRICS_V2_KERNEL_COUNTERS_PER_CPU))
+			{
+				// We set the num events for that CPU.
+				set_u64_monotonic_kernel_counter(pos, cnt_map.n_evts, METRICS_V2_KERNEL_COUNTERS_PER_CPU);
+				snprintf(g_state.stats[pos].name, METRIC_NAME_MAX, N_EVENTS_PER_CPU_PREFIX"%d", index);
+				pos++;
+
+				// We set the drops for that CPU.
+				set_u64_monotonic_kernel_counter(pos, cnt_map.n_drops_buffer + cnt_map.n_drops_max_event_size, METRICS_V2_KERNEL_COUNTERS_PER_CPU);
+				snprintf(g_state.stats[pos].name, METRIC_NAME_MAX, N_DROPS_PER_CPU_PREFIX"%d", index);
+				pos++;
+			}
+		}
+		offset = pos;
+	}
+
+	/* KERNEL COUNTERS PER CPU STATS 
+	 * The following `if` handle the case in which we want to get the metrics per CPU but not the global ones.
+	 * It is an unsual case but at the moment we support it.
+	 */
+	if ((flags & METRICS_V2_KERNEL_COUNTERS_PER_CPU) && !(flags & METRICS_V2_KERNEL_COUNTERS))
```

Comment:
[EARLY FEEDBACK]

Actually, i'm managing also this weird case in which we have the per-CPU stats enabled but not the global ones... I cannot think of a real use case for it so I'm not sure we want to keep it. WDYT? @FedeDP @incertum

Reply:
I'd say that if KERNEL_COUNTERS are disabled, METRICS_V2_KERNEL_COUNTERS_PER_CPU must be disabled too!

Reply:
I see 3 options:
1. handling the 2 stats separately (so looping 2 times among CPUs if both are enabled)
2. handling the 2 stats together so `METRICS_V2_KERNEL_COUNTERS_PER_CPU` can be enabled only if `METRICS_V2_KERNEL_COUNTERS` is enabled. We create a dependecy
3. like case 1 but we have a duplicated logic that allow us to loop just once if both metrics are enabled (implemented in this PR)

Reply:
I'd go with 2, easier and expected since both metrics flag share same prefix.

Reply:
We can also say that if only `METRICS_V2_KERNEL_COUNTERS_PER_CPU` is passed, we silently enable `METRICS_V2_KERNEL_COUNTERS` too.

Reply:
> We can also say that if only METRICS_V2_KERNEL_COUNTERS_PER_CPU is passed, we silently enable METRICS_V2_KERNEL_COUNTERS too.

Great idea, i will go for it!